api/src/org/apache/cloudstack/api/command/user/vm/UpdateVMCmd.java

```diff
@@ -136,13 +136,13 @@ public String getInstanceName() {
         return instanceName;
     }
 
-    public Map getDetails() {
+    public Map<String, String> getDetails() {
```

Comment:
how about taking the opportunity to change this to 
    public Map<String, String> getDetails() {
        if (this.details == null || this.details.isEmpty()) {
            return null;
        }
        return details;
    }
Is there any reason to pass through the crazy construct of getting the pointer to the first element and casting it back to a map? It seems like very unsafe. It becomes even more unsafe when using an instance of a generic type.